tools/wpt/browser.py

```diff
@@ -41,14 +41,22 @@ def handle_remove_readonly(func, path, exc):
         raise
 
 
+def get_ext(filename):
+    """Get the extension from a filename with special handling for .tar.foo"""
+    name, ext = os.path.splitext(filename)
+    if name.endswith(".tar"):
+        ext = ".tar%s" % ext
+    return ext
+
+
 class Browser(object):
     __metaclass__ = ABCMeta
 
     def __init__(self, logger):
         self.logger = logger
 
     @abstractmethod
-    def download(self, dest=None, channel=None):
+    def download(self, dest=None, channel=None, rename=None):
```

Comment:
IIUC, `dest` is the destination for the installed/extracted browser binary, whereas `rename` is the target filename for the downloaded tarball? We should have a docstring for this...